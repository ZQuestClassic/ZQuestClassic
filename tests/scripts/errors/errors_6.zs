class Car {
	int a;
	int a;

	void blah(){}
	void blah(){}
}
class Car {
}

int a = 1;
int a = 1;

void blah(){}
void blah(){}
