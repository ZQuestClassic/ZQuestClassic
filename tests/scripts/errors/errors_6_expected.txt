stderr:

Compiling 'errors_6.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
errors_6.zs Line 12 @ Columns 5-6 - Error S008: There is already a variable with name a in this scope.

10    
11    int a = 1;
12    int a = 1;
          ^


Failure!
Compile finished with exit code '8' (compiled with errors)

stdout:

{
  "success": false,
  "code": 8,
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 11,
          "character": 4
        },
        "end": {
          "line": 11,
          "character": 5
        }
      },
      "severity": 1,
      "message": "S008: There is already a variable with name a in this scope."
    }
  ],
  "metadata": {
    "currentFileSymbols": 10,
    "symbols": 0,
    "identifiers": 0,
    "elided": true
  }
}
